src/include/sof/trace/trace.h

```diff
@@ -435,42 +439,59 @@ struct tr_ctx {
 
 /* tracing from infrastructure part */
 
-#define tr_err(ctx, fmt, ...) \
-	trace_error_with_ids(_TRACE_INV_CLASS, ctx, \
-			     _TRACE_INV_ID, _TRACE_INV_ID, fmt, ##__VA_ARGS__)
-
 #define tr_err_atomic(ctx, fmt, ...) \
 	trace_error_atomic_with_ids(_TRACE_INV_CLASS, ctx, \
 				    _TRACE_INV_ID, _TRACE_INV_ID, \
 				    fmt, ##__VA_ARGS__)
 
-#define tr_warn(ctx, fmt, ...) \
-	trace_warn_with_ids(_TRACE_INV_CLASS, ctx, \
-			    _TRACE_INV_ID, _TRACE_INV_ID, fmt, ##__VA_ARGS__)
-
 #define tr_warn_atomic(ctx, fmt, ...) \
 	trace_warn_atomic_with_ids(_TRACE_INV_CLASS, ctx, \
 				   _TRACE_INV_ID, _TRACE_INV_ID, \
 				   fmt, ##__VA_ARGS__)
 
-#define tr_info(ctx, fmt, ...) \
-	trace_event_with_ids(_TRACE_INV_CLASS, ctx, \
-			     _TRACE_INV_ID, _TRACE_INV_ID, fmt, ##__VA_ARGS__)
-
 #define tr_info_atomic(ctx, fmt, ...) \
 	trace_event_atomic_with_ids(_TRACE_INV_CLASS, ctx, \
 				    _TRACE_INV_ID, _TRACE_INV_ID, \
 				    fmt, ##__VA_ARGS__)
 
+#define tr_dbg_atomic(ctx, fmt, ...) \
+	tracev_event_atomic_with_ids(_TRACE_INV_CLASS, ctx, \
+				     _TRACE_INV_ID, _TRACE_INV_ID, \
+				     fmt, ##__VA_ARGS__)
+
+#ifdef __ZEPHYR__
+
+#define tr_err(ctx, fmt, ...) LOG_ERR(fmt, ##__VA_ARGS__)
+
+#define tr_warn(ctx, fmt, ...) LOG_WRN(fmt, ##__VA_ARGS__)
+
+#define tr_info(ctx, fmt, ...) LOG_INF(fmt, ##__VA_ARGS__)
+
+#define tr_dbg(ctx, fmt, ...) LOG_DBG(fmt, ##__VA_ARGS__)
+
+#else
+
+#define LOG_MODULE_REGISTER(ctx, level)
+#define LOG_MODULE_DECLARE(ctx, level)
+
```

Comment:
We also need to define LOG() here for xtos users.

Reply:
No, I am not invent a new macro, and use it to wrap LOG_XXX(zephyr) and tr_xxx(XTOS). I just reuse those tr_xxx/comp_xxx/pipe_xxx macro, when sof build with zephyr, they are replaced with LOG_XXX, otherwise, they are replaced with xtos logging utilities.